android/src/toga_android/window.py

```diff
@@ -137,8 +146,68 @@ def get_visible(self):
     # Window state
     ######################################################################
 
-    def set_full_screen(self, is_full_screen):
-        self.interface.factory.not_implemented("Window.set_full_screen()")
+    def get_window_state(self):
+        # `window.state` is called in `_close()`, which itself is
+        # sometimes called during certain stages when the app
+        # attribute may not exist. In such cases, return NORMAL.
+        if getattr(self, "app", None) is None:
+            return WindowState.NORMAL
+        decor_view = self.app.native.getWindow().getDecorView()
+        system_ui_flags = decor_view.getSystemUiVisibility()
+        if system_ui_flags & (
+            decor_view.SYSTEM_UI_FLAG_FULLSCREEN
+            | decor_view.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+            | decor_view.SYSTEM_UI_FLAG_IMMERSIVE
+        ):
+            if self._in_presentation_mode:
+                return WindowState.PRESENTATION
+            else:
+                return WindowState.FULLSCREEN
+        return WindowState.NORMAL
+
+    def set_window_state(self, state):
+        current_state = self.get_window_state()
+        decor_view = self.app.native.getWindow().getDecorView()
+
+        if current_state == state:
+            return
```

Comment:
This can't be done at the core level because we can't reliably determine the current state. Backends like Cocoa and GTK use non-blocking state APIs, meaning the state might be in transition when checked. As a result, we could end up comparing against an outdated state.

Reply:
I have modified get_window_state() and moved same state checking to the core. I have also moved same state transition tests to the core.

Reply:
No, not really, since the Android API is blocking in nature and the internal API `set_window_state()` is not called internally anywhere on the implementation side. 

It is only required on the non-blocking API backends like cocoa, as the internal API `_apply_state()` is called internally on the implementation side at different places.

So, I have removed it from the Android backend.